src/components/ProductCard.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useEffect, useState } from "react";
+import { Transition } from "@headlessui/react";
+import Image from "next/image";
+import ProudctNameAndEdition from "./ProductNameAndEdition";
+import ProductColourButton from "./ProductColourButton";
+
+type Content = {
+  colours: string[];
+  image: string[];
+  index: number;
+  name: string;
+  edition?: string;
+  price: string;
+};
+
+export default function ProductCard({
+  content,
+  index,
+}: {
+  content: Content;
+  index: number;
+}) {
```

Comment:
Not particularlly fond of inline typing like this since you've already created a type up top, could simplify by turning this into a type so that's it's a little easier to follow

Reply:
removed index and use content.index inside of return.